cdap-api-common/src/main/java/co/cask/cdap/api/data/format/RecordFormat.java

```diff
@@ -77,7 +77,11 @@ public void initialize(FormatSpecification formatSpecification) throws Unsupport
     }
     desiredSchema = desiredSchema == null ? getDefaultSchema() : desiredSchema;
     if (desiredSchema == null) {
-      throw new UnsupportedTypeException("A schema must be provided to this format.");
+      String msg = "A schema must be provided to the format: ";
+      if (formatSpecification != null) {
```

Comment:
I took a brief look, and it seems like the only places where it can be null is where it is read from ``Properties`` in the ``StreamSerDe`` and other places in the REST API where it is deserialized using ``GSON``.